nautobot/extras/forms/mixins.py

```diff
@@ -107,6 +107,13 @@ def _append_customfield_fields(self):
             # Annotate the field in the list of CustomField form fields
             self.custom_fields.append(field_name)
 
+        try:
+            if self.Meta.fields == "__all__":
```

Comment:
Not sure this check is necessary, TBH - I see you're testing this below, but is there ever actually a valid use case for trying to present `_custom_field_data` as a form field?

Reply:
I cannot think of a valid use case, and I approached it that way at first, but I figured if there was one I couldn't think of, then not considering this extra check would not follow the principal of least astonishment.

Reply:
Fair point. That said it's a bit surprising itself that this removal happens in `_append_customfield_fields()` since nothing about that function name suggests that it's going to remove anything - maybe this logic should be located directly in `__init__` above instead?